lib/utils/pdf_to_images.rb

```diff
@@ -31,6 +29,7 @@ def extract_pages
 
       raise Discourse::InvalidParameters.new("Failed to download PDF") if pdf_path.nil?
 
+      temp_dir = Dir.mktmpdir("discourse-pdf-#{SecureRandom.hex(8)}")
```

Comment:
Now that this is moved into the `begin..ensure` context, we need to change `FileUtils.rm_rf(temp_dir)` to `FileUtils.rm_rf(temp_dir) if temp_dir`

Reply:
good point will fix